### DISCUSSION TITLE
Store both access_token and id_token, not mutually exclusive

### DISCUSSION THREAD

**Discussion by radarcg** [nuxt-community/auth-module#1628] [Ideas & Feature Requests]
This is a request as follow up to the closed issue, https://github.com/nuxt-community/auth-module/issues/366. 

I seem to be running into a similar issue using the identity provider support for Cognito User Pools. According to the documentation, /userinfo must send an access_token: https://docs.amazonaws.cn/en_us/cognito/latest/developerguide/cognito-user-pools-oidc-flow.html

However, to authenticate a user with an integrated API Gateway using the Cognito User Pool Authorizer, the id_token must be used: https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-enable-cognito-user-pool.html

So this is yet another use case where the auth library needs to store both the access_token and the id_token.

**Comment by kilakewe** [comment]
I'm with you here on this one. It's been a problem for the past year for me.

**Comment by yoshwata** [comment]
I am facing a similar problem. I would like to use the ID token for Hasura authorization. Also, the authentication process requires access to the Auth0 API, which requires an access token.

https://auth0.com/docs/security/tokens

If multiple responseTypes are set in `nuxt.config.json` like `'token id_token'`, only the access token is stored in the local storage. In this situation I cannot authenticate requests to Hasura.

I can get ID token by specifying only `id_token`, but the login process will not complete.
![スクリーンショット_2022-01-06_8_03_34](https://user-images.githubusercontent.com/32473622/148316104-c2ea1af6-02b8-4d0f-b836-85737e130173.png)

I want a mechanism that can obtain both an access token for login and an ID token for using the application.
